subscription-service/src/main/kotlin/com/egm/stellio/subscription/web/SubscriptionHandler.kt

```diff
@@ -72,43 +81,35 @@ class SubscriptionHandler(
     @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE])
     suspend fun getSubscriptions(
         @RequestHeader httpHeaders: HttpHeaders,
-        @RequestParam(required = false, defaultValue = "1") page: Int,
-        @RequestParam(required = false, defaultValue = SUBSCRIPTION_QUERY_PAGING_LIMIT.toString()) limit: Int,
+        @RequestParam params: MultiValueMap<String, String>,
         @RequestParam options: Optional<String>
     ): ResponseEntity<*> {
+        val page = params.getFirst(PAGE_PARAM_ID)?.toIntOrNull() ?: 1
+        val limit = params.getFirst(LIMIT_PARAM_ID)?.toIntOrNull() ?: paginationDefaultLimit
         val includeSysAttrs = options.filter { it.contains("sysAttrs") }.isPresent
         val contextLink = getContextFromLinkHeaderOrDefault(httpHeaders)
         val mediaType = getApplicableMediaType(httpHeaders)
         if (limit <= 0 || page <= 0)
             return ResponseEntity.status(HttpStatus.BAD_REQUEST).contentType(MediaType.APPLICATION_JSON)
                 .body(BadRequestDataResponse("Page number and Limit must be greater than zero"))
 
+        if (limit > paginationMaxLimit)
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).contentType(MediaType.APPLICATION_JSON)
+                .body(BadRequestDataResponse("Maximum limit is $paginationMaxLimit"))
```

Comment:
you could have something more user friendly ... "You asked for X results, but the supported maximum limit is Y"